README.md

```diff
@@ -157,6 +159,50 @@ xtf.foo.v2.version=1.0.3
 
 Retrieving an instance with this metadata: `Produts.resolve("product");`
 
+#### Using `TestCaseContext` to get name of currently running test case
+
+If `junit.jupiter.extensions.autodetection.enabled=true` then JUnit 5 extension `cz.xtf.core.context.TestCaseContextExtension` is
+automatically registered. It
+sets name of currently running test case into `TestCaseContext` before `@BeforeAll` of test case is called.
+
+Following code then can be used to retrieve the name of currently running test case in:
+```
+String testCase = TestCaseContext.getRunningTestCaseName()
+```
+
+#### Automatic creation of namespace(s)
+
+XTF allows to automatically manage creation of testing namespace which is defined by `xtf.openshift.namespace` property. This 
+namespace is created before any test case is started. 
+
+This feature requires to have XTF JUnit5 `cz.xtf.junit5.listeners.ProjectCreator` extension enabled. This can be done by adding
+`cz.xtf.junit5.listeners.ProjectCreator` line into files:
+```
+src/test/resources/META-INF/services/org.junit.jupiter.api.extension.Extension
+src/test/resources/META-INF/services/org.junit.platform.launcher.PostDiscoveryFilter
+src/test/resources/META-INF/services/org.junit.platform.launcher.TestExecutionListener
+```
+
+#### Run test cases in separate namespaces using `xtf.openshift.namespace.per.testcase` property 
+
+You can enable running each test case in separate namespace by setting `xtf.openshift.namespace.per.testcase=true`. 
+
+Namespace names follow pattern: "`${xtf.openshift.namespace}`-TestCaseName". 
+For example for `xtf.openshift.namespace=testnamespace` and test case `org.test.SmokeTest` it will be `testnamespace-SmokeTest`.
+
+You can limit the length of created namespace by `xtf.openshift.namespace.per.testcase.length.limit` property. By default it's `25` chars. If limit is breached then
+test case name in namespace name is hashed to hold the limit. So namespace name would like `testnamespace-s623jd6332`
+
+**Warning - Limitations**
+
+When enabling this feature in your project, **you may need to replace `OpenShiftConfig.getNamespace()` with`NamespaceManager.getNamespace()`. Check method's javadoc to understand difference.**
```

Comment:
Nice to have. Is it possible to provide links to: 
- `OpenShiftConfig.getNamespace()` 
- `NamespaceManager.getNamespace()`

So reader can check javadoc as reads.

Reply:
Right, fixed.

Reply:
Fixed. Resolving.